### PR TITLE
Fix crash on `<textarea>` without end tag in `vue/html-indent` rule

### LINKED ISSUE #1752: vue/html-indent errors while using <textarea>
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.6.0
- **eslint-plugin-vue version:** 8.2.0
- **Node version:** 16.13.1
- **Operating System:** Ubuntu 20.04.3 

**Please show your full configuration:**

```js
module.exports = {
	"env": {
		"commonjs": true,
	},
	"extends": [
		"eslint:recommended",
		"plugin:vue/vue3-strongly-recommended"
	],
	"parserOptions": {
		"ecmaVersion": 13,
		"sourceType": "module"
	},
	"plugins": [
		"vue"
	],
	"rules": {
		"vue/html-indent": ["error", "tab"]
	}
};
```

**What did you do?**	

```vue
<template>
	<textarea class="red">
		abc
</template>
```

**What did you expect to happen?**
Linting errors about missing a closing textarea tag

**What actually happened?**

```
Oops! Something went wrong! :(

ESLint: 8.6.0

TypeError: Cannot read properties of null (reading 'range')
Occurred while linting /home/cheeze/repos/test-app/App.vue:1
Rule: "vue/html-indent"
    at TokenStore.getTokensBetween (/home/cheeze/repos/test-app/node_modules/vue-eslint-parser/index.js:4240:107)
    at setPreformattedTokens (/home/cheeze/repos/test-app/node_modules/eslint-plugin-vue/lib/utils/indent-common.js:314:36)
    at EventEmitter.VElement (/home/cheeze/repos/test-app/node_modules/eslint-plugin-vue/lib/utils/indent-common.js:971:9)
    at EventEmitter.emit (node:events:402:35)
    at NodeEventGenerator.applySelector (/home/cheeze/repos/test-app/node_modules/vue-eslint-parser/index.js:3826:26)
    at NodeEventGenerator.applySelectors (/home/cheeze/repos/test-app/node_modules/vue-eslint-parser/index.js:3840:22)
    at NodeEventGenerator.enterNode (/home/cheeze/repos/test-app/node_modules/vue-eslint-parser/index.js:3848:14)
    at traverse (/home/cheeze/repos/test-app/node_modules/vue-eslint-parser/index.js:117:13)
    at traverse (/home/cheeze/repos/test-app/node_modules/vue-eslint-parser/index.js:124:21)
    at traverseNodes (/home/cheeze/repos/test-app/node_modules/vue-eslint-parser/index.js:135:5)
```

**Repository to reproduce this issue**

[https://github.com/cheeze2000/eslint-vue-error](https://github.com/cheeze2000/eslint-vue-error)
To reproduce the error, `npm install` and `npm run lint`.

### PR DESCRIPTION
**Pull request by ota-meshi**
fixes #1752

### CODE REVIEW

File: lib/utils/indent-common.js [hunk 1]
```diff
@@ -311,11 +311,9 @@ module.exports.defineVisitor = function create(
...
           token.type === 'HTMLComment')
     }
-    for (const token of tokenStore.getTokensBetween(
-      node.startTag,
-      endToken,
-      option
-    )) {
+    for (const token of endToken
+      ? tokenStore.getTokensBetween(node.startTag, endToken, option)
+      : tokenStore.getTokensAfter(node.startTag, option)) {
```
Comment (FloEdelmann) [inline lib/utils/indent-common.js:316]: I'd extract this into a separate variable `contentTokens` to make the for loop more readable. Also, please rename `option` to `options` while you're at it.

**Review by FloEdelmann — COMMENTED** [review]
Looks good, apart from a minor readability improvement.

I guess that now `vue/no-parsing-error` complains about this missing end tag?

File: lib/utils/indent-common.js (hunk 1 above)
Comment (ota-meshi) [reply lib/utils/indent-common.js]: Thank you for your review! I changed it.

**Comment by ota-meshi** [comment]
> I guess that now vue/no-parsing-error complains about this missing end tag?

`vue/no-parsing-error` rule does not report an error. Users can find the error using the `vue/html-end-tags` rule.

https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJxtjMEKwyAQBX8l7DlLoMcQ+iW9iD5aixpxVwmE/HttDyGHHmcGZie7OtBMiyLmYBT3RxqGTpuaAtNpmS6JRio1QGjeqVVMdo0RSdn5Aqu+9dVt/JW3bFwFws0UOa3Y4rOyQGv+k18aAyM5VvP82uP4ANkCOHw=

**Review by FloEdelmann — APPROVED** [review]
LGTM :+1:

**Pull request merged**